### PR TITLE
Issue #299: Add SuppressionPatchFilter introductions and descriptions in the README

### PR DESCRIPTION
**Pull request by HuGanghui**
Issue #299: Add SuppressionPatchFilter introductions and descriptions in the README

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,11 +1,39 @@
...
+
+#### Examples
+
+For example, the following configuration fragment directs the Checker to use a SuppressionPatchFilter with patch file config/defaultContext.patch:
+```xml
+<module name="com.puppycrawl.tools.checkstyle.filters.SuppressionPatchFilter">
+    <property name="file" value="config/defaultContext.patch" />
+    <property name="strategy" value="newline" />
+</module>
+```
```
Comment (rdiachenko) [inline README.md:30]: Please add example for `neverSuppressedChecks`

File: README.md [hunk 2]
```diff
@@ -1,11 +1,39 @@
...
+ | strategy              | Control suppression scope that you need. If `startegy` is set to `newline`, it only accepts audit events for Check violations whose line number belong to added lines in patch file. `patchedline` will accept added/changed lines. | [String](https://checkstyle.sourceforge.io/property_types.html#String) | newline       |
+ | neverSuppressedChecks | String has user defined Checks to never suppress if files are touched, split by comma. This property is useful for Checks that place violation on whole file or on not all (first/last) occurrence of cause/violated code. | [String](https://checkstyle.sourceforge.io/property_types.html#String) | null          |
+
+#### Examples
+
+For example, the following configuration fragment directs the Checker to use a SuppressionPatchFilter with patch file config/defaultContext.patch:
+```xml
+<module name="com.puppycrawl.tools.checkstyle.filters.SuppressionPatchFilter">
+    <property name="file" value="config/defaultContext.patch" />
```
Comment (rdiachenko) [inline README.md:27]: `value="file.patch"`

Comment (HuGanghui) [reply README.md]: done

File: README.md (hunk 1 above)
Comment (HuGanghui) [reply README.md:30]: done

File: README.md [hunk 3]
```diff
@@ -1,11 +1,51 @@
...
+    <property name="file" value="config/file.patch" />
+    <property name="strategy" value="newline" />
+</module>
+```
+
+the following configuration fragment direts the Checker to use a SuppressionPatchFilter 
+with patch file config/file.patch, whose strategy is `patchedline`,and never suppress 
+checks `EmptyBlock` and `HiddenField`:
+```xml
+<module name="com.puppycrawl.tools.checkstyle.filters.SuppressionPatchXpathFilter">
```
Comment (rdiachenko) [inline README.md:36]: Wrong filter

**Review by rdiachenko — CHANGES_REQUESTED** [review]

File: README.md (hunk 3 above)
Comment (HuGanghui) [reply README.md]: done

**Review by rdiachenko — APPROVED** [review]
LGTM

**Pull request merged**